### PR TITLE
chore: Parrellelize running tests

### PR DESCRIPTION
**Pull request by devaradhanm**
#### Description of changes
This change has brought down the build time from 10-12 to 7 min.
I have split the build in to two jobs.
1.) Job 1 - builds, lint & publishes artifacts & run 1/4th of tests.
2.) Job 2  - builds & runs the remaining 4/5th of tests.

I have split the tests like this to get approximately same amount of time taken in both the jobs.

Build before this change - https://dev.azure.com/accessibility-insights/Accessibility%20Insights%20Service/_build/results?buildId=7510&view=results
Build after this change - https://dev.azure.com/accessibility-insights/Accessibility%20Insights%20Service/_build/results?buildId=7515&view=results

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (lamaks) [inline azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml]: can rename file name to more specific or remove `-some-`

File: azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      displayName: check for copyright headers
+
+    - template: ./build-steps.yaml
+
+    - script: |
+          yarn lint:check
+      displayName: run linter
+
+    - template: ./test-steps.yaml
+      parameters: { totalSlices: '${{ parameters.totalSlices }}', slicesToRun: '${{ parameters.slicesToRun }}' }
```
Comment (lamaks) [inline azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml:21]: what is `slice`?  can we have some generic message?

File: azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+              web-api/dist/**/*
+              web-workers/dist/**/*
+              functional-tests/dist/**/*
+              health-client/dist/**/*
+              cli/drop/*
+              !**/node_modules/**/*
+              !**/.vscode/**/*
+              !**/*.md
+          TargetFolder: '$(Build.ArtifactStagingDirectory)'
+      displayName: 'Copy Files to staging directory'
```
Comment (lamaks) [inline azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml:46]: NIT files

File: azure-pipeline/remaining-tests-job.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+steps:
+    - template: ./install-dependencies-steps.yaml
+    - template: ./build-steps.yaml
+    - template: ./test-steps.yaml
+      parameters: { totalSlices: '${{ parameters.totalSlices }}', slicesToRun: '${{ parameters.slicesToRun }}' }
```
Comment (lamaks) [inline azure-pipeline/remaining-tests-job.yaml:7]: the same for `slice` as above

File: azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml (hunk 1 above)
Comment (devaradhanm) [reply azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml:1]: can u suggest a name? I am running 1/4th of tests in the 1st job & remaining tests in the second job

File: azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml (hunk 2 above)
Comment (devaradhanm) [reply azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml:21]: I am slicing the tests in to n numbers & then run the project tests that belong to the slice numbers specified.

Comment (devaradhanm) [reply azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml:21]: any suggestion on naming?

File: azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml (hunk 1 above)
Comment (lamaks) [reply azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml:1]: remove some

File: azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml (hunk 2 above)
Comment (lamaks) [reply azure-pipeline/build-artifacts-and-some-tests-job-steps.yaml]: never mind

**Pull request merged**